devito/ir/equations/equation.py

```diff
@@ -154,23 +154,24 @@ class ClusterizedEq(Eq, IREq, FrozenExpr):
     """
 
     def __new__(cls, *args, **kwargs):
-        if len(args) == 3:
+        if len(args) == 2:
```

Comment:
Readability: overloading different meanings based on the number of arguments.  maybe two `classmethod`s could be preferable here?  e.g. `from_lhs_rhs_stamp` `from_expr_ispace_dspace`?

Reply:
but you would still need to use ``if-then-else`` to invoke one or the other? or what am I missing?

Reply:
Well, I meant that you'd call one of the classmethods to create an object, rather than using __new__ (or perhaps __new__ is for one of the patterns only).  But I guess that's not great either.  I don't know.

Reply:
ah! well no that's not possible because `__new__` is called via SymPy too, not just internally by Devito (in fact, I think it's never called explicitly in Devito)